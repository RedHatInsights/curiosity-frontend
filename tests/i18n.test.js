const get = require('lodash/get');
const { GettextExtractor, JsExtractors } = require('gettext-extractor');
const enLocales = require('../public/locales/en');

const textExtractor = () => {
  const extractor = new GettextExtractor();
  extractor
    .createJsParser([
      JsExtractors.callExpression(['t', '[this].t'], {
        arguments: {
          text: 0,
          context: 1
        }
      })
    ])
    .parseFilesGlob('./src/components/**/*!(.test|.spec).@(js|jsx)');

  return extractor;
};

describe('i18n locale', () => {
  const getText = textExtractor();

  it('should generate a predictable pot output snapshot', () => {
    expect(getText.getPotString()).toMatchSnapshot('pot output');
  });

  it('should have locale keys that exist in the default language JSON', () => {
    const messages = (getText.getMessages && getText.getMessages()) || [];
    const missingKeys = [];

    messages.forEach(value => {
      const keyCheck = (value && value.text) || '';

      if (keyCheck && !get(enLocales, keyCheck, null)) {
        missingKeys.push(keyCheck);
      }
    });

    expect(missingKeys).toMatchSnapshot('missing locale keys');
  });
});
