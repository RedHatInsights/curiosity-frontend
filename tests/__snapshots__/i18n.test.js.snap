// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`i18n locale should generate a predictable pot output snapshot: pot output 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Content-Type: text/plain; charset=UTF-8\\\\n\\"

#: src/common/graphHelpers.js:46
msgid \\"curiosity-graph.tooltipLabel\\"
msgstr \\"\\"

#: src/common/graphHelpers.js:61
msgid \\"curiosity-graph.tooltipPreviousLabelDaily\\"
msgstr \\"\\"

#: src/common/graphHelpers.js:54
msgid \\"curiosity-graph.tooltipPreviousLabelMonthly\\"
msgstr \\"\\"

#: src/common/graphHelpers.js:57
msgid \\"curiosity-graph.tooltipPreviousLabelQuarterly\\"
msgstr \\"\\"

#: src/common/graphHelpers.js:51
msgid \\"curiosity-graph.tooltipPreviousLabelWeekly\\"
msgstr \\"\\"

#: src/components/rhelGraphCard/rhelGraphCard.js:79
msgctxt \\"CPU socket usage\\"
msgid \\"curiosity-graph.heading\\"
msgstr \\"\\"

#: src/components/rhelGraphCard/rhelGraphCardTypes.js:7
msgctxt \\"Daily\\"
msgid \\"curiosity-graph.dropdownDaily\\"
msgstr \\"\\"

#: src/components/rhelGraphCard/rhelGraphCardTypes.js:9
msgctxt \\"Monthly\\"
msgid \\"curiosity-graph.dropdownMonthly\\"
msgstr \\"\\"

#: src/components/rhelGraphCard/rhelGraphCardTypes.js:10
msgctxt \\"Quarterly\\"
msgid \\"curiosity-graph.dropdownQuarterly\\"
msgstr \\"\\"

#: src/components/rhelGraphCard/rhelGraphCard.js:82
#: src/components/rhelGraphCard/rhelGraphCard.js:86
msgctxt \\"Select date range\\"
msgid \\"curiosity-graph.dropdownPlaceholder\\"
msgstr \\"\\"

#: src/components/rhelGraphCard/rhelGraphCardTypes.js:8
msgctxt \\"Weekly\\"
msgid \\"curiosity-graph.dropdownWeekly\\"
msgstr \\"\\"
"
`;

exports[`i18n locale should have locale keys that exist in the default language JSON: missing locale keys 1`] = `Array []`;
